tests/unit/test_charm.py

```diff
@@ -387,3 +387,99 @@ def test_datasource(
 
     harness.update_relation_data(mysql_relation_id, "mysql-k8s", relation_data)
     assert harness.charm._datasource() == expected_output
+
+
+@pytest.mark.parametrize(
+    "files, expected",
+    [
+        (
+            [
```

Comment:
Should these be tuples since they should not be modified in the test?

Reply:
Also applicable for the other lists in parametrize

Reply:
Fixed